src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,68 @@
+import { Completed, Filters } from '../../types/Filters';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  filterOptions: Filters;
+  onFilterChange: (key: keyof Filters, value: string | Completed) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  filterOptions,
+  onFilterChange,
+}) => {
+  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedValue = event.target.value as Completed;
+
+    onFilterChange('completedType', selectedValue);
+  };
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.target.value.toLowerCase();
```

Comment:
Converting the search query to lowercase here is fine, but ensure that the filtering logic in the parent component also handles case insensitivity correctly, which it seems to do based on the provided code.